kv/etcd/store_test.go

```diff
@@ -272,14 +273,28 @@ func TestWatchLastVersion(t *testing.T) {
 	require.NoError(t, err)
 	require.Nil(t, w.Get())
 
+	var wg sync.WaitGroup
 	lastVersion := 100
+	errC := make(chan error, lastVersion)
+	wg.Add(lastVersion)
 	go func() {
 		for i := 1; i <= lastVersion; i++ {
 			_, err := store.Set("foo", genProto(fmt.Sprintf("bar%d", i)))
-			require.NoError(t, err)
+			if err != nil {
+				errC <- err
```

Comment:
instead of asserting no err here, we should atomic inc the number of errs from all the set and compare
value.Version() == lastVersion - atomic.LoadInt32(numErrs) in the next for loop, otherwise this test will still be flaky since there will be errors from etcd cluster.